pkg/kcp/provider/gcp/iprange/checkGcpOperation.go

```diff
@@ -27,8 +27,17 @@ func checkGcpOperation(ctx context.Context, st composed.State) (error, context.C
 		ipRange.Status.State == client.DeletePsaConnection {
 		op, err := state.serviceNetworkingClient.GetServiceNetworkingOperation(ctx, opName)
 		if err != nil {
-			state.AddErrorCondition(ctx, v1beta1.ReasonGcpError, err)
-			return composed.LogErrorAndReturn(err, "Error getting operation from GCP", composed.StopWithRequeue, nil)
+			return composed.UpdateStatus(ipRange).
+				SetCondition(metav1.Condition{
+					Type:    v1beta1.ConditionTypeError,
+					Status:  metav1.ConditionTrue,
+					Reason:  v1beta1.ReasonGcpError,
+					Message: err.Error(),
+				}).
+				RemoveConditionIfReasonMatched(v1beta1.ConditionTypeError, v1beta1.ReasonGcpError).
+				SuccessError(composed.StopWithRequeue).
```

Comment:
Minor: I think onUpdateSuccess is more readable than SuccessError just because of the "Error" in the name is mostly about internal implementation and therefore confusing. This comment applies to the similar calls in this PR.